File: Two Pointers/Java/RemoveDuplicatesFromSorted.java
/*

https://leetcode.com/problems/remove-duplicates-from-sorted-list/submissions/

Given the head of a sorted linked list, delete all duplicates such that each element appears only once. 

Return the linked list sorted as well.


Input: head = [1,1,2]
Output: [1,2]


Input: head = [1,1,2,3,3]
Output: [1,2,3]


Constraints:

The number of nodes in the list is in the range [0, 300].
-100 <= Node.val <= 100
The list is guaranteed to be sorted in ascending order.
*/


public class RemoveDuplicatesFromSorted{

    public ListNode deleteDuplicates(ListNode head){

        ListNode current = head;
        while(current != null && current.next != null){
            if(current.next.val != current.val){ 
                current = current.next;
            }else{
                current.next = current.next.next;
            }

        }

        return head;

    }

    private class ListNode{

        int val;
        ListNode next;

        ListNode(){}
        ListNode(int val){ this.val = val;}
        ListNode(int val, ListNode.next){
            this.val = val;
            this.next = next;
        }

    }






}

File: Two Pointers/Python/RemoveDuplicatesFromSorted.py
"""
Given the head of a sorted linked list, delete all duplicates such that each element appears only once. Return the linked list sorted as well.
"""

class ListNode:
     def __init__(self, val=0, next=None):
         self.val = val
         self.next = next

class Solution:
    def deleteDuplicates(self, head):
        if(head == None):
            return head
        vals = []
        new_head = ListNode()
        while(head):
            if(head.val in vals):
                vals = vals
            else:
                vals.append(head.val)
            head = head.next
        
        new_head.val = vals[0]
        temp = new_head
        for i in vals[1:]:
            print(i)
            new_node = ListNode(i)
            temp.next = new_node
            temp = new_node
        return new_head